analyzer/src/main/kotlin/curation/ClearlyDefinedPackageCurationProvider.kt

```diff
@@ -110,47 +115,51 @@ class ClearlyDefinedPackageCurationProvider(
                 }
 
                 else -> {
                     e.showStackTrace()
-
-                    log.warn {
-                        "Querying ClearlyDefined curation for '${pkgId.toCoordinates()}' failed: " +
-                                e.collectMessagesAsString()
-                    }
+                    log.warn { "Querying curations failed: ${e.collectMessagesAsString()}" }
                 }
             }
-        }.getOrNull() ?: return emptyList()
+        }.getOrNull() ?: return emptyMap()
 
-        val declaredLicenseParsed = curation.licensed?.declared?.let { declaredLicense ->
-            // Only take curations of good quality (i.e. those not using deprecated identifiers) and in particular none
-            // that contain "OTHER" as a license, also see https://github.com/clearlydefined/curated-data/issues/7836.
-            runCatching { declaredLicense.toSpdx(SpdxExpression.Strictness.ALLOW_CURRENT) }.getOrNull()
-        }
+        val curations = mutableMapOf<Identifier, MutableList<PackageCuration>>()
 
-        val sourceLocation = curation.described?.sourceLocation.toArtifactOrVcs()
+        contributedCurations.forEach { (_, contributions) ->
+            contributions.curations.forEach inner@{ (coordinates, curation) ->
+                val pkgId = coordinatesToIds[coordinates] ?: return@inner
 
-        val pkgCuration = PackageCuration(
-            id = pkgId,
-            data = PackageCurationData(
-                concludedLicense = declaredLicenseParsed,
-                homepageUrl = curation.described?.projectWebsite?.toString(),
-                sourceArtifact = sourceLocation as? RemoteArtifact,
-                vcs = sourceLocation as? VcsInfoCurationData,
-                comment = "Provided by ClearlyDefined."
-            )
-        )
+                val declaredLicenseParsed = curation.licensed?.declared?.let { declaredLicense ->
+                    // Only take curations of good quality (i.e. those not using deprecated identifiers) and in
+                    // particular none that contain "OTHER" as a license, also see
+                    // https://github.com/clearlydefined/curated-data/issues/7836.
+                    runCatching { declaredLicense.toSpdx(SpdxExpression.Strictness.ALLOW_CURRENT) }.getOrNull()
+                }
+
+                val sourceLocation = curation.described?.sourceLocation.toArtifactOrVcs()
+
+                curations.getOrPut(pkgId) { mutableListOf() } += PackageCuration(
+                    id = pkgId,
+                    data = PackageCurationData(
+                        concludedLicense = declaredLicenseParsed,
+                        homepageUrl = curation.described?.projectWebsite?.toString(),
+                        sourceArtifact = sourceLocation as? RemoteArtifact,
+                        vcs = sourceLocation as? VcsInfoCurationData,
+                        comment = "Provided by ClearlyDefined."
+                    )
```

Comment:
That should already be the case, or?

Reply:
Is it? Let's say there is only a curation for the declared license which we discard because it does not satisfy `ALLOW_CURRENT` it looks to me like we would add an empty curation.

Reply:
Ah, now I believe I get what you mean: As an improvement to also the existing code, you're suggesting to check all entries of `PackageCurationData` to be non-null before adding a curation, right? I could add a new commit for that, though I doubt we'll often (if ever) run into that case.

Reply:
Yes, I just wanted to mention it as I noticed it during the review. I'll leave it up to you if you want to fix it.

Reply:
I see, you already did :)

Reply:
I've added a commit.